src/Searchbar.vue

```diff
@@ -15,38 +15,90 @@ export default {
   },
   computed: {
     primitiveData() {
+      function allMatch(info, regexes) {
+        let matched = true;
+        regexes.forEach((regex) => {
+          if (!regex.test(info)) {
+            matched = false;
+          }
+        });
+        return matched;
+      }
+
       if (!this.data) {
         return undefined;
       }
       const matches = [];
-      const regex = new RegExp(this.value, 'i');
+      const regexes = this.value.split(' ').map(keyword => new RegExp(keyword, 'i'));
       this.data.forEach((entry) => {
         const { headings, src, title } = entry;
         const keywords = entry.keywords || '';
         let hasMatchingHeading = false;
+
         Object.entries(headings).forEach(([id, text]) => {
-          if (regex.test(text)) {
-            hasMatchingHeading = true;
-            matches.push({
-              heading: { id, text },
-              keywords,
-              src,
-              title,
-            });
-          }
-        });
-        if (!hasMatchingHeading) {
-          if (regex.test(title) || regex.test(keywords)) {
+            const info = [title].concat(keywords).concat(text).join(' ');
```

Comment:
This is not correct. If `title` and `keywords` match, all headings will be matched.

Reply:
This is for those that doesn't have headings

Reply:
If there's no heading, then this code would not run at all?

Reply:
then if title and keywords match, shouldn't all headings match?

Reply:
prof want one entry for full page, then entries for each headings in this case.